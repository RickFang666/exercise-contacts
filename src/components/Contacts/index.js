import { arrayOf, object } from 'prop-types';
import styled from 'styled-components';
import Contact from '../Contact';

const List = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
`;

const Contacts = ({ contacts }) => (
  <List>
    { contacts.map(contact => <Contact key={contact.id} {...contact} />) }
  </List>
);

Contacts.propTypes = {
  contacts: arrayOf(object).isRequired,
};

export default Contacts;
