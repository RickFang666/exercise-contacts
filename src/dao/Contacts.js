import url from 'url';
import fetch from 'isomorphic-fetch';

export default async function fetchContacts(options) {
  const apiUrl = url.format({
    protocol: 'http',
    hostname: 'localhost',
    port: 3000,
    pathname: '/api/contacts',
    query: options,
  });

  const result = await fetch(apiUrl);

  return result.json();
}
