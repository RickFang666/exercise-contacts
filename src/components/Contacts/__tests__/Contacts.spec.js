import Contacts from '../';

it('renders correctly', () => {
  const contacts = [
    {
      id: 'test-id-1',
      name: 'test-name-1',
    },
    {
      id: 'test-id-2',
      name: 'test-name-2',
    },
    {
      id: 'test-id-3',
      name: 'test-name-3',
    },
  ];
  const result = shallow(<Contacts contacts={contacts} />);
  expect(result).toMatchSnapshot();
});
