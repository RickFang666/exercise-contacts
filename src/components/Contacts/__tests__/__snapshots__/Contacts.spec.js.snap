// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="contacts"
>
  <Contact
    id="test-id-1"
    name="test-name-1"
  />
  <Contact
    id="test-id-2"
    name="test-name-2"
  />
  <Contact
    id="test-id-3"
    name="test-name-3"
  />
</ul>
`;
